samples/ai/src/index.ts

```diff
@@ -0,0 +1,136 @@
+import { App } from '@microsoft/teams.apps';
+import { DevtoolsPlugin } from '@microsoft/teams.dev';
+// :snippet-start: ai-imports
+import { ChatPrompt } from '@microsoft/teams.ai';
+import { OpenAIChatModel } from '@microsoft/teams.openai';
+// :snippet-end:
+import { MessageActivity } from '@microsoft/teams.api';
+import { handleStatefulConversation } from './stateful-prompts';
+import { handleGetWeatherToolCalling, handlePokemonToolCalling } from './tool-calling';
+
+const app = new App({
+  plugins: [new DevtoolsPlugin()],
+});
+
+const model = new OpenAIChatModel({
+  apiKey: process.env.AZURE_OPENAI_API_KEY || process.env.OPENAI_API_KEY,
+  endpoint: process.env.AZURE_OPENAI_ENDPOINT,
+  apiVersion: process.env.AZURE_OPENAI_API_VERSION,
+  model: process.env.AZURE_OPENAI_MODEL_DEPLOYMENT_NAME!,
+});
+
+// Handle "hi" message
+// :snippet-start: simple-chat
+app.on('message', async ({ send, activity, next }) => {
+  // :remove-start:
+  if (activity.text.toLowerCase() !== 'hi') {
+    await next();
+    return;
+  }
+  // :remove-end:
+  const model = new OpenAIChatModel({
+    apiKey: process.env.AZURE_OPENAI_API_KEY || process.env.OPENAI_API_KEY,
+    endpoint: process.env.AZURE_OPENAI_ENDPOINT,
+    apiVersion: process.env.AZURE_OPENAI_API_VERSION,
+    model: process.env.AZURE_OPENAI_MODEL_DEPLOYMENT_NAME!,
+  });
+
+  const prompt = new ChatPrompt({
+    instructions: 'You are a friendly assistant who talks like a pirate',
+    model,
+  });
+
+  const response = await prompt.send(activity.text);
+  if (response.content) {
+    const activity = new MessageActivity(response.content).addAiGenerated();
+    await send(activity);
+    // Ahoy, matey! 🏴‍☠️ How be ye doin' this fine day on th' high seas? What can this ol’ salty sea dog help ye with? 🚢☠️
+  }
+});
+// :snippet-end:
+
+/**
+ *
+ * @param commandStr the user-facing command string
+ * @param commandName the name of the command
+ * @returns
+ */
+const extractCommandAndQueryForCommand =
+  <TCommandName extends string>(commandStr: string, commandName: TCommandName) =>
+  (text: string): { commandName: TCommandName; query: string } | null => {
+    const parts = text.split(' ');
+    const command = parts.at(0);
+    if (!command) {
+      return null;
+    }
+    if (command === commandStr) {
+      return { commandName: commandName, query: parts.slice(1).join(' ') };
+    }
+    return null;
+  };
+
+const pokemonCommand = extractCommandAndQueryForCommand('pokemon', 'pokemon-tool-calling');
+const weatherCommand = extractCommandAndQueryForCommand('weather', 'get-weather-tool-calling');
+const streamCommand = extractCommandAndQueryForCommand('stream', 'streaming-chat');
+
+// Handle "<supported-command> <query>" message
+app.on('message', async ({ send, activity, next }) => {
+  const commandAndQuery = pokemonCommand(activity.text) || weatherCommand(activity.text);
+  if (!commandAndQuery) {
+    await next();
```

Comment:
@aacebo do you know why I must await this? It really matters for streaming to work. I'm guessing without awaiting the stream.close() runs too early? Atleast that's what my debugging shows. 

If that's the case, should we change the types for next to be `Promise<...>` instead of `Promise<...> | ...`

Reply:
no need to await, this works with streaming when not awaited.